### PR TITLE
Fixes #28214 - lock all input fields

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

**Comment by ares** [comment]
This fixes a bug when locked template does not lock (disable) all input form elements

before
before

after
after

**Comment by xprazak2** [comment]
The 'Value Type' and 'Resource Type' fields are now disabled, however their styling is a bit different when compared with 'Input Type' field, perhaps that is caused because 'Input Type' is without select2. Text is not grayed out, there is hover effect on field border and cursor does not change. 

Are we ok with that?

**Comment by ares** [comment]
That's a good catch, however I'm not sure how to fix it. Do you have some recommendation?

**Comment by xprazak2** [comment]
I managed some visual tweaks with https://github.com/xprazak2/foreman/commit/60cb8b80838fbb4bfe40f925e88513403da890e9

**Comment by ares** [comment]
Works well, thanks @xprazak2 I squashed that in

**Review by xprazak2 — APPROVED** [review]

File: app/views/template_inputs/_form.html.erb [hunk 1]
```diff
@@ -4,8 +4,8 @@
...
       <%= selectable_f f, :input_type, template_input_types_options(@template.acceptable_template_input_types), {}, :class => 'input_type_selector without_select2', :disabled => @template.locked? %>
-      <%= selectable_f f, :value_type, [[_('Plain'), 'plain'], [_("Search"), 'search'], [_('Date'), 'date']], {}, {'data-item': f.index || f.object.id, class: 'select-value-type', onchange: "tfm.templateInputs.inputValueOnchange(this)"}  %>
-      <%= selectable_f f, :resource_type, Permission.resources_with_translations, {}, {wrapper_class: "#{hide_resource_type_input(f.object)} form-group resource-type-#{f.index || f.object.id}"} %>
+      <%= selectable_f f, :value_type, [[_('Plain'), 'plain'], [_("Search"), 'search'], [_('Date'), 'date']], {}, {'data-item': f.index || f.object.id, class: 'select-value-type', onchange: "tfm.templateInputs.inputValueOnchange(this)", :disabled => @template.locked? } %>
+      <%= selectable_f f, :resource_type, Permission.resources_with_translations, {}, {wrapper_class: "#{hide_resource_type_input(f.object)} form-group resource-type-#{f.index || f.object.id}", :disabled => @template.locked? } %>
```
Comment (tbrisker) [inline app/views/template_inputs/_form.html.erb:8]: Should these just change to use without_select2 class like the `input_type` field, so we don't have different styling? I am concerned the added styles might break the styling on a different page somewhere.

**Comment by ares** [comment]
@tbrisker I did what you asked, however the styling looks wrong again, see the screenshot - font color is not grey and the cursor icon is missing

Screenshot_20200522_084211

I don't know what exactly causes that but I think the suggested CSS was actually correct for all disabled select2 on every page for consistency. Please let me know your thoughts, I kept changes in separate commit so I can easily revert to the original patch

**Comment by ares** [comment]
for some reason, I can't reopen the PR

**Pull request closed without merging**